libs/Communicator/Receiver.hpp

```diff
@@ -94,6 +97,30 @@ namespace communicator_lib
         /// @brief Private async function holding the network search process
         void wait(void);
 
+        /// @brief Extract the communication header from the transfered data
+        /// @param data The transfered data
+        /// @return A vector of two unsigned short which represent the data header
+        std::vector<unsigned short> getDataHeader(void *data);
+
+        /// @brief Trait the data for the dataType 10
+        /// @param dataContent The transfered data (Client + Data)
+        void dataTraitmentType10(Message dataContent);
+
+        /// @brief Trait the data for the dataType 20
+        /// @param dataContent The transfered data (Client + Data)
+        void dataTraitmentType20(Message dataContent);
+
+        /// @brief Trait the data for the dataType 21
+        /// @param dataContent The transfered data (Client + Data)
+        void dataTraitmentType21(Message dataContent);
+
+        /// @brief Trait the data for the dataType 30
+        /// @param dataContent The transfered data (Client + Data)
+        void dataTraitmentType30(Message dataContent);
```

Comment:
> trait the data

It's not precise enough. It seems these function "add message" somewhere, it's more precise than "trait data".